GPflowOpt/domain.py

```diff
@@ -46,6 +46,12 @@ def __add__(self, other):
         assert isinstance(other, Domain)
         return Domain(self._parameters + other._parameters)
 
+    def __radd__(self, other):
+        if other == 0:
+            return self
+
+        return other + self
+
```

Comment:
Great, that seems to work. Glad to get rid of radd.

With sum() radd is triggered when summing an one-element list:
domain = sum([ContinuousParameter('x', 0, 1)])

While np.sum solves this without having to use radd, I added a test for the above just in case